macleann/enchiridion-client
# Cypress Component Testing

# Cypress Component Testing!

After poring over the wonderful Cypress docs, I've been able to implement individual component testing for nearly all necessary components. The only remaining components I've yet to implement are those in `src/components/search` and `src/components/seasons`. Next, I think I'll move on to e2e testing? I need to do more research to determine how common it is to perform both component and e2e testing.

As expected with component testing, I needed to add some `id` attributes to many `<div>` tags as needed. I also needed to fix some functionality with my drag and drop implementation, which ultimately simplified my code. Per Cypress' documentation, I've also implemented the testing step into my GitHub Actions file that may need to be changed later.

Of course, I also had to create the many test component files, which I decided to nest within the relevant directories containing the components they're testing. I also created some fixture files that I just copied from valid responses from my server.

## Type of change


# How to Test?

Make sure you've followed all installation and setup instructions in the README.

```
git fetch origin nm-testing
git checkout nm-testing
ENV_VARIABLE=VALUE npx cypress open
```

- [ ] Please note that all environment variables will need to be included in the last line above. I couldn't figure out how, for component testing, how to embed the variables within my cypress configuration so that they were set at component run time. For e2e, I anticipate this being much easier.
- [ ] Using Cypress' UI, run all tests and they should all pass

# Checklist:
